File: api/src/main/java/com/FireSale/api/dto/UserDTO.java
package com.FireSale.api.dto;

import com.FireSale.api.model.Gender;
import com.FireSale.api.model.Role;
import lombok.Data;

import java.time.LocalDate;
import java.util.Set;

@Data
public class UserDTO extends BaseDTO {
    private String email;
    private String displayName;
    private String firstName;
    private String lastName;
    private LocalDate dateOfBirth;
    private Gender gender;
    private Role role;
    private Set<String> claims;
}

File: client/FireSale/src/app/users/models/userDto.ts
import { BaseDto } from 'src/app/shared/baseDto';
import { AddressDto } from './addressDto';

export interface UserDto extends BaseDto {
    email: string;
    gender: string;
    firstName: string;
    lastName: string;
    address: AddressDto;
    shippingAddress: AddressDto;
    dateOfBirth: Date;
    displayName: string;
    password: string;
    passwordVerify: string;
    authData: string;
    avatar: string;
}
